### PR TITLE
Configure kube pod resources for workers/syncs

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Implements resource managements for worker/sync kube containers from #3638

## How
have default (and editable) resource allocations for launched pods in the production Kustomize overlay
have default (and ui/api configurable) resource allocations per connector (which means they'd be stored in our configs) in a specific connection

## Recommended reading order
1. `airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java`
2. `airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessFactory.java`
3. `kube/resources/scheduler.yaml`
4. `docker-compose.yaml`
5. the rest

### CODE REVIEW

File: kube/resources/resource-defaults.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v1
+kind: LimitRange
```
Comment (jrhizor) [inline kube/resources/resource-defaults.yaml:2]: See comment on other PR.

**Review by jrhizor — COMMENTED** [review]
There are likely cases where we want to have unbounded resource allocation, like on local docker for testing. Generally like where this is going though. Will review in more depth once it's out of draft.

File: kube/resources/resource-defaults.yaml (hunk 1 above)
Comment (ChristopheDuong) [reply kube/resources/resource-defaults.yaml:2]: this was not supposed to be in this PR 😁

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 2]
```diff
@@ -580,4 +600,19 @@ public int exitValue() {
...
+    if (resourceRequirements != null) {
+      final Map<String, Quantity> resourceMap = new HashMap<>();
+      // if null then use unbounded resource allocation
+      if (!com.google.common.base.Strings.isNullOrEmpty(resourceRequirements.getCpu())) {
+        resourceMap.put("cpu", Quantity.parse(resourceRequirements.getCpu()));
+      }
+      if (!com.google.common.base.Strings.isNullOrEmpty(resourceRequirements.getMemory())) {
+        resourceMap.put("memory", Quantity.parse(resourceRequirements.getMemory()));
+      }
+      return new ResourceRequirementsBuilder().withLimits(resourceMap);
```
Comment (ChristopheDuong) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:613]: On this PR, I am just exposing the "limits" (not the requests parts) for the moment, I'm not sure if that's enough as a start? or should both be configurable?

File: airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java [hunk 3]
```diff
@@ -328,8 +342,14 @@ public KubePodProcess(ApiClient officialClient,
         .withMountPath(TERMINATION_DIR)
         .build();
 
-    Container init = getInit(usesStdin, List.of(pipeVolumeMount, configVolumeMount));
-    Container main = getMain(image, usesStdin, entrypoint, List.of(pipeVolumeMount, configVolumeMount, terminationVolumeMount), args);
+    Container init = getInit(usesStdin, List.of(pipeVolumeMount, configVolumeMount));;
```
Comment (jrhizor) [inline airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:345]:
```suggestion
    Container init = getInit(usesStdin, List.of(pipeVolumeMount, configVolumeMount));
```

Comment (jrhizor) [reply airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java:345]: In the future we may want to manually set the container requirements for the init / non-main containers. They should all be really lightweight.

File: docker-compose.yaml [hunk 4]
```diff
@@ -28,6 +28,11 @@ services:
       - POSTGRES_PASSWORD=${DATABASE_PASSWORD}
     volumes:
       - db:/var/lib/postgresql/data
+    deploy:
+      resources:
+        limits:
+          cpus: "0.25"
+          memory: 4G
```
[the remark this answers is not in the document]
Comment (jrhizor) [reply docker-compose.yaml:35]: I don't think we should have resource limits for these by default. For users trying out Airbyte on their laptop I don't want them to run into problems being able to run this when they're below 4g of free memory.

The default for launched containers feels like it should be "no limit" as well, which can then be overridden to a specific value that can become the default, like `RESOURCE_CPU_REQUEST` if set.

**Review by jrhizor — APPROVED** [review]
Generally looks good.

I think the main change that I would like is for the default behavior for Kube and Docker to be "no resource limits", both at the core and connector level. This should be configurable for prod instances.

@cgardens can you also take a quick glance at this PR, mostly the API changes? My sense is that for cloud we'll likely need to limit the range of assignable values (at first we can require them to just be the same as the default).

**Pull request merged**